heron/proto/ckptmgr.proto

```diff
@@ -195,6 +218,9 @@ message SaveInstanceStateRequest {
   // Information about the instance whose state this is
   required heron.proto.system.Instance instance = 1;
   required InstanceStateCheckpoint checkpoint = 2;
+  // Stmgrs need to send component metadata so that Checkpoint
+  // Mgrs know how to repartition if needed.
+  optional CheckpointComponentMetadata metadata = 3;
```

Comment:
does this need to be passed to cjptmgr? Doesn't ckptmgr know this info

Reply:
yeah. this info will be passed to ckptmgr, so that it knows if repartition is needed and how to repartition.

Reply:
my comment was more that when you change the parallelism, wouldnt the ckptmgr be restarted also and thus it too knows the topology structure?

Reply:
chptmgr is stateless and it doesn't have the physical plan, so it doesn't know the parallelism by itself.

Reply:
Checked the communication between stmgr and ckptmgr, it is a valid option to pass the topology structure information during the registration to simplify the requests.

We don't need to make the decision for now about where the structure information should be passed. Therefore I am going to remove this field for now to keep it simple.

Thanks again for the comments!